internal/metrics/json.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+type LscpuEntry struct {
+	Field    string       `json:"field"`
+	Data     string       `json:"data"`
+	Children []LscpuEntry `json:"children,omitempty"`
+}
+
+type Lscpu struct {
+	Lscpu []LscpuEntry `json:"lscpu"`
+}
+
+func parseJSON(r io.Reader, structPtr interface{}) (interface{}, error) {
```

Comment:
you should start using `any` which is way more recent than this code, true, but for empty interface, this is more idiomatic in the meaning.

I don’t really like the `Ptr` suffix to the struct and as the function is rather small, I would align with the Go standard library with just `v`: https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/encoding/json/decode.go;l=97

Btw, did you notice they converted their code to any? :p

Reply:
I think this code is still 1.13 and `any` isn't supported in this context yet! Would you like me to maybe convert it?

Reply:
yes, feel free to bump the Go version!

Reply:
Done!